app/src/main/java/com/futurice/freesound/feature/search/SearchActivity.java

```diff
@@ -160,4 +167,15 @@ private void addSearchFragment() {
                                    .commit();
     }
 
+    @Override
+    public void onDestroy() {
+        searchSnackbar.dismissSnackbar();
+        super.onDestroy();
+    }
+    public void showSnackbar(CharSequence charSequence){
```

Comment:
I was looking at this before and I think that there are better approaches than having public method on the `Activity`. It might be a indirect, but the `Fragment` should be calling a public method on a shared `ViewModel` instance. The `Activity` would subscribe to this corresponding signal and show the `Snackbar` accordingly.